sway-lsp/src/utils/keyword_docs.rs

```diff
@@ -0,0 +1,850 @@
+            /// struct's list of fields can be omitted, it's usually kept for convenience in adding and
+            /// removing fields down the line.
+            mod struct_keyword {}
+        };
+
+        let enum_keyword: ItemMod = parse_quote! {
+            /// A type that can be any one of several variants.
+            ///
+            /// Enums in Sway are similar to those of other compiled languages like C, but have important
+            /// differences that make them considerably more powerful. What Sway calls enums are more commonly
+            /// known as [Algebraic Data Types][ADT] if you're coming from a functional programming background.
+            /// The important detail is that each enum variant can have data to go along with it.
+            ///
+            /// ```sway
+            /// # struct Coord;
+            /// enum SimpleEnum {
+            ///     FirstVariant,
+            ///     SecondVariant,
+            ///     ThirdVariant,
+            /// }
+            ///
+            /// enum Location {
+            ///     Unknown,
+            ///     Anonymous,
+            ///     Known(Coord),
+            /// }
+            ///
+            /// enum ComplexEnum {
+            ///     Nothing,
+            ///     Something(u32),
+            ///     LotsOfThings {
+            ///         usual_struct_stuff: bool,
+            ///         blah: u8,
+            ///     }
+            /// }
+            ///
+            /// enum EmptyEnum { }
+            /// ```
+            ///
+            /// The first enum shown is the usual kind of enum you'd find in a C-style language. The second
+            /// shows off a hypothetical example of something storing location data, with `Coord` being any
+            /// other type that's needed, for example a struct. The third example demonstrates the kind of
+            /// data a variant can store, ranging from nothing, to a tuple, to an anonymous struct.
+            ///
+            /// Instantiating enum variants involves explicitly using the enum's name as its namespace,
+            /// followed by one of its variants. `SimpleEnum::SecondVariant` would be an example from above.
+            /// When data follows along with a variant, such as with sway's built-in [`Option`] type, the data
+            /// is added as the type describes, for example `Option::Some(123)`. The same follows with
+            /// struct-like variants, with things looking like `ComplexEnum::LotsOfThings { usual_struct_stuff:
+            /// true, blah: 245, }`. Empty Enums are similar to [`!`] in that they cannot be
+            /// instantiated at all, and are used mainly to mess with the type system in interesting ways.
+            ///
+            /// [ADT]: https://en.wikipedia.org/wiki/Algebraic_data_type
+            mod enum_keyword {}
+        };
+
+        let self_keyword: ItemMod = parse_quote! {
```

Comment:
I see the hover for `self` when it's a function parameter, but not when it's used inside of a function block (e.g. `self.bar`)

<img width="436" alt="image" src="https://user-images.githubusercontent.com/47993817/216196834-2cc89423-6a43-4b33-b87b-2c8043a907cd.png">

Reply:
Hmm interesting. I'll open an issue and follow up on this.

Reply:
opened #3966